include/irr/asset/IMeshPacker.h

```diff
@@ -21,13 +18,96 @@ template <typename MeshBufferType>
 class IMeshPacker
 {
 public:
-    virtual std::optional<std::pair<MeshBufferType* ,DrawElementsIndirectCommand_t>> packMeshes(const core::vector<MeshBufferType*>& meshBuffers) = 0;
+    struct AllocationParams
+    {
+        size_t indexBuffSupportedSizeInBytes                    = 2147483648ull; /*  2GB*/
+        size_t vertexBuffSupportedSizeInBytes                   = 4294967296ull; /*  4GB*/
+        size_t perInstanceVertexBuffSupportedSizeInBytes        = 33554432ull;   /* 32MB*/
+        size_t MDIDataSizeInBytes                               = 2147483648ull; /*  2GB*/
+        size_t vertexBufferMinAllocSize                         = 256ull;        /* 256B*/
+        size_t indexBufferMinAllocSize                          = 512ull;        /* 512B*/
+        size_t perInstanceVertexBufferMinAllocSize              = 32ull;         /*  32B*/
+    };
+
+    template <typename MeshBufferType>
+    struct PackedMeshBuffer
+    {
+        core::smart_refctd_ptr<MeshBufferType> packedMeshBuffer = nullptr;
+    };
+
+    struct PackedMeshBufferData
+    {
+        const size_t mdiParameterByteOffset; // add to `CCPUMeshPacker::getMultiDrawIndirectBuffer()->getPointer() to get `DrawElementsIndirectCommand_t` address
+        const uint32_t mdiParameterCount;
+    };
+
+public:
+
+    IMeshPacker(const SVertexInputParams& preDefinedLayout, const AllocationParams& allocParams,uint16_t maxIndexCountPerMDIData = std::numeric_limits<uint16_t>::max())
```

Comment:
good default for `maxIndexCountPerMDIData` is 1024 or 256

Reply:
btw a static method in the CCPUMeshPacker to help the user figure out how many packers ( `vector<SVertexInputParams> preDefinedLayout`) we need from a list (iterator pair) of ICPUMeshBuffers would be handy!